lib/hydra/works/models/concerns/generic_file/characterization.rb

```diff
@@ -0,0 +1,120 @@
+      property :last_modified, predicate: RDF::Vocab::EBUCore.dateModified
+      property :fits_version, predicate: RDF::Vocab::PREMIS.hasCreatingApplicationVersion
+      property :exif_version, predicate: EXIF.exifVersion
+      property :original_checksum, predicate: RDF::Vocab::PREMIS.hasMessageDigest
+      # property :mime_type, delegate_to: 'original_file'
+      property :mime_type, predicate: RDF::Vocab::EBUCore.hasMimeType, multiple: false do |index|
+        index.as :stored_searchable
+      end
+
+      # Parser config is :term => :property, and is consulted before defaulting to a
+      # property that matches the term.  Required when term and property aren't the same.
+      class_attribute :parser_config
+      self.parser_config = { exif_tool_version: :exif_version,
+                             file_mime_type: :mime_type }
+
+      class_attribute :parser_class
+      self.parser_class = FitsDatastream
+
+      # Extract the metadata from a content file and store it as properties.
+      # The target properties must already exist.
+      # @param [ActiveFedora::File] file to characterize
+      def characterize(file = original_file)
+        # Run fits and get xml string
+        extracted_md = extract_metadata(file)
+        # Parse xml string to get fits datastream terms and values
+        terms = parse_metadata(extracted_md)
+        # Store terms for which there are properties mapped.
+        store_metadata(terms)
+      end
+
+      protected
+
+      def extract_metadata(file = original_file)
+        return unless file.has_content?
+        Hydra::FileCharacterization.characterize(file.content, file.original_name, :fits) do |cfg|
+          cfg[:fits] = Hydra::Derivatives.fits_path
+        end
+      end
+
+      # Use OM to parse metadata
+      def parse_metadata(metadata)
+        fdata = parser_class.new
+        fdata.ng_xml = metadata if metadata.present?
+        characterization_terms(fdata)
+      end
+
+      # Get terms and values
+      # This is using the names of the proxy terms; not the name of what the proxy points to.
+      def characterization_terms(datastream)
+        h = {}
+        datastream.class.terminology.terms.each_pair do |key, target|
+          # a key is a proxy if it's target responds to proxied_term
+          next unless target.respond_to? :proxied_term
+          begin
+            h[key] = datastream.send(key)
+          rescue NoMethodError
+            next
+          end
+        end
+        h.delete_if { |_k, v| v.empty? }
+      end
+
+      # Assign values of the instance properties from the metadata mapping :prop => val
+      def store_metadata(terms)
+        # Go through each term.
+        terms.each_pair do |term, value|
+          # Figure out property to use or nil
+          property = property_for(term)
+          next if property.nil?
+          # Array-ify the value to avoid a conditional here
+          value = [value] unless value.is_a? Array
```

Comment:
Yup! Pass it an array and it returns the original array. Pass it a non-array and it array-ifies it safely.

(`Array.wrap` is even safer but I believe that's from ActiveSupport.)